crates/subspace-gateway/src/piece_getter.rs

```diff
@@ -64,6 +65,22 @@ where
 
         Ok(None)
     }
+
+    async fn get_pieces<'a, PieceIndices>(
+        &'a self,
+        piece_indices: PieceIndices,
+    ) -> anyhow::Result<
+        Box<dyn Stream<Item = (PieceIndex, anyhow::Result<Option<Piece>>)> + Send + Unpin + 'a>,
+    >
+    where
+        PieceIndices: IntoIterator<Item = PieceIndex, IntoIter: Send> + Send + 'a,
+    {
+        Ok(Box::new(
+            self.get_from_cache(piece_indices)
+                .await
+                .map(|(index, maybe_piece)| (index, Ok(maybe_piece))),
```

Comment:
Probably worth having a TODO that this is supposed to eventually reach out to archival storage too and do reconstruction if necessary. Cache is what should be used basically all the time, but cold storage fallback is still necessary for completeness.